frontend/ui/uimanager.lua

```diff
@@ -714,7 +707,9 @@ function UIManager:handleInput()
 
     -- delegate input_event to handler
     if input_event then
-        self:_resetAutoSuspendTimer()
+        if input_event.handler ~= "onInputError" then
+            self:broadcastEvent(Event:new("InputEvent"))
```

Comment:
Yes, I have the same feeling. But the significant disadvantage of hook is that UIManager will always keep a reference of the object and eventually blocks GC from recycling.
I am still considering a better solution to solve both problems.